src/parseTools.mjs

```diff
@@ -995,11 +995,15 @@ function from64Expr(x) {
   return `Number(${x})`;
 }
 
+// Converts a value to BigInt if building for wasm64, with both 64-bit pointers
+// and 64-bit memory. Used for indices into the memory tables, for example.
 function toIndexType(x) {
   if (MEMORY64 == 1) return `BigInt(${x})`;
   return x;
 }
 
+// Converts a value to BigInt if building for wasm64, regardless of whether the
+// memory is 32- or 64-bit. Used for passing 64-bit values to the JS API.
```

Comment:
I wonder if we should convert all the comments in this file to jsdoc?  (doesn't have to be part of this PR).

Reply:
How about "Used for passing pointer-width values to native code.   Since pointers are presented as Number in JS and BigInt in wasm we need to convert before passing them".

Reply:
> I wonder if we should convert all the comments in this file to jsdoc?

It's probably worth doing but there are 1000+ lines to scroll through.

I updated to comment with your suggestion.